engine/utils.hpp

```diff
@@ -98,21 +98,22 @@ static inline std::string string_view_2_string(const StringView &src) {
   return std::string(src.data(), src.size());
 }
 
-static inline int compare_string_view(const StringView &src,
-                                      const StringView &target) {
-  auto size = std::min(src.size(), target.size());
+static inline int compare_string_view(const char *src, size_t src_len,
```

Comment:
why change this api?

Reply:
Considering C API,   user can't pass `string` or `string_view`. So `char` * is more suitable.

Reply:
for c api, you can still combine two char* and sizes to stringview in c.cc

Reply:
How to implement ?  based on the current desgin

Reply:
why I can't change this api?

Reply:
I renamed this function name. If you want to hold before function, I keep it.

Reply:
void KVDKSetSortedCompareFunc(KVDKEngine *engine, const char *collection,
                              int (*key_comp_func)(const char *, size_t,
                                                   const char *, size_t),
                              int (*val_comp_func)(const char *, size_t,
                                                   const char *, size_t)) {
auto key_view_comp = [](const StringView &src, const StringView &target){
    return key_comp_func(src.data(), src.size(), target.data(), target.size());
};
auto val_view_comp = /* same as above */;
engine->rep->SetSortedCompareFunc(pmem::obj::string_view(collection),
                                    key_view_comp, val_view_comp);
}

Reply:
there is no need to do.  char* to string_view  or string_view to char *  is the same.

Reply:
change api to (char*, size, char*, size) makes code ugly with no benefits.